changes.d/6117.feat.md

```diff
@@ -0,0 +1 @@
+Make `workflow/share/cycle/<cycle>` available to jobs as `$CYLC_TASK_SHARE_CYCLE_DIR`.
```

Comment:
We should communicate that we are also creating this directory as it might not have existed before for non-rose users.